source/logging/LogQueue.cpp

```diff
@@ -25,7 +25,7 @@ std::unique_ptr<LogMessage> LogQueue::getNextLog()
     unique_lock<mutex> readLock(queueLock);
     while (logQueue.empty() && !isShutdown)
     {
-        newLogNotifier.wait(readLock);
+        newLogNotifier.wait_for(readLock, chrono::seconds(1));
```

Comment:
I think this change will increase the CPU consumption since the while condition is validated after `x` seconds in loop.

Reply:
You are right, this change will certainly cost additional CPU cycles every x seconds. The current fix is the simplest change we can make right now, which can be fine if our overall CPU consumption is still low. However, if this is too big of an issue, then we will have to figure out another way to fix the bug.

Reply:
In that case lets test everything (CPU consumption) before pushing this change in mainline.

Reply:
Setting the timeout period to 1 millisecond consumes 1% CPU on my ec2 instance. Anything over 100 milliseconds has
<img width="568" alt="Screen Shot 2022-04-18 at 5 04 59 PM" src="https://user-images.githubusercontent.com/100961047/163895985-63fdfb51-3d84-466a-ae13-279406222f75.png">
 0.0 - 0.1% CPU usage. Of course we cannot assume anything about customer cpu performance but I think something like 200ms should be safe for now

Reply:
Perhaps in the future we can optimize the code further to achieve even less resource consumption but the logger would need to be reworked to ensure correctness at the same time.